type_description_interfaces/msg/FieldType.msg

```diff
@@ -0,0 +1,157 @@
+uint8 FIELD_TYPE_INT32_BOUNDED_SEQUENCE = 102
+uint8 FIELD_TYPE_UINT32_BOUNDED_SEQUENCE = 103
+uint8 FIELD_TYPE_INT64_BOUNDED_SEQUENCE = 104
+uint8 FIELD_TYPE_UINT64_BOUNDED_SEQUENCE = 105
+uint8 FIELD_TYPE_FLOAT_BOUNDED_SEQUENCE = 106
+uint8 FIELD_TYPE_DOUBLE_BOUNDED_SEQUENCE = 107
+uint8 FIELD_TYPE_LONG_DOUBLE_BOUNDED_SEQUENCE = 108
+uint8 FIELD_TYPE_CHAR_BOUNDED_SEQUENCE = 109
+uint8 FIELD_TYPE_WCHAR_BOUNDED_SEQUENCE = 110
+uint8 FIELD_TYPE_BOOLEAN_BOUNDED_SEQUENCE = 111
+uint8 FIELD_TYPE_BYTE_BOUNDED_SEQUENCE = 112
+uint8 FIELD_TYPE_STRING_BOUNDED_SEQUENCE = 113
+uint8 FIELD_TYPE_WSTRING_BOUNDED_SEQUENCE = 114
+uint8 FIELD_TYPE_FIXED_STRING_BOUNDED_SEQUENCE = 115
+uint8 FIELD_TYPE_FIXED_WSTRING_BOUNDED_SEQUENCE = 116
+uint8 FIELD_TYPE_BOUNDED_STRING_BOUNDED_SEQUENCE = 117
+uint8 FIELD_TYPE_BOUNDED_WSTRING_BOUNDED_SEQUENCE = 118
+
+# Unbounded Sequence Types
+uint8 FIELD_TYPE_NESTED_TYPE_UNBOUNDED_SEQUENCE = 145
+uint8 FIELD_TYPE_INT8_UNBOUNDED_SEQUENCE = 146
+uint8 FIELD_TYPE_UINT8_UNBOUNDED_SEQUENCE = 147
+uint8 FIELD_TYPE_INT16_UNBOUNDED_SEQUENCE = 148
+uint8 FIELD_TYPE_UINT16_UNBOUNDED_SEQUENCE = 149
+uint8 FIELD_TYPE_INT32_UNBOUNDED_SEQUENCE = 150
+uint8 FIELD_TYPE_UINT32_UNBOUNDED_SEQUENCE = 151
+uint8 FIELD_TYPE_INT64_UNBOUNDED_SEQUENCE = 152
+uint8 FIELD_TYPE_UINT64_UNBOUNDED_SEQUENCE = 153
+uint8 FIELD_TYPE_FLOAT_UNBOUNDED_SEQUENCE = 154
+uint8 FIELD_TYPE_DOUBLE_UNBOUNDED_SEQUENCE = 155
+uint8 FIELD_TYPE_LONG_DOUBLE_UNBOUNDED_SEQUENCE = 156
+uint8 FIELD_TYPE_CHAR_UNBOUNDED_SEQUENCE = 157
+uint8 FIELD_TYPE_WCHAR_UNBOUNDED_SEQUENCE = 158
+uint8 FIELD_TYPE_BOOLEAN_UNBOUNDED_SEQUENCE = 159
+uint8 FIELD_TYPE_BYTE_UNBOUNDED_SEQUENCE = 160
+uint8 FIELD_TYPE_STRING_UNBOUNDED_SEQUENCE = 161
+uint8 FIELD_TYPE_WSTRING_UNBOUNDED_SEQUENCE = 162
+uint8 FIELD_TYPE_FIXED_STRING_UNBOUNDED_SEQUENCE = 163
+uint8 FIELD_TYPE_FIXED_WSTRING_UNBOUNDED_SEQUENCE = 164
+uint8 FIELD_TYPE_BOUNDED_STRING_UNBOUNDED_SEQUENCE = 165
+uint8 FIELD_TYPE_BOUNDED_WSTRING_UNBOUNDED_SEQUENCE = 166
+
+# Identifying number for the type of the field, using one of the above constants.
+uint8 type_id
+
+# Only used when the type is an array or a bounded sequence.
+# In the case of an array, this is the fixed length of the array.
+# In the case of a bounded sequence, this is the maximum size of the sequence.
+# In all other cases this field is unused.
+uint64 length
+
+# Only used when the type is a fixed or bounded string/wstring, or a array/sequence of those.
+# In the case of a fixed string/wstring, it is the fixed length of the string.
+# In the case of a bounded string/wstring, it is the maximum length of the string.
+# In the case of an array/sequence of fixed string/wstring, it is the fixed length of the strings.
+# In the case of an array/sequence of bounded string/wstring, it is the maximum length of the strings.
+# It is not currently possible to have different string lengths per element in the array/sequences.
+uint64 string_length
```

Comment:
If, at some point in the future, bounded sequences of bounded sequences are supported, it might be better to make this term more generic:

```suggestion
uint64 element_capacity
```

Reply:
I'm not sure about this one.  @wjwwood could you maybe comment about this?

Reply:
Since that's not the case now I wouldn't call it element length yet. I think I'm the meantime it would be confusing. Also I'm not sure we'd ever support sequence of sequence.